### ISSUE TITLE
BUG: negative value in min needs report

### ISSUE THREAD

**Issue by Charlotte-Morgan** [inasafe/inasafe#4039]

**Comment by Gustry** [comment]
@Charlotte-Morgan Can we have more inputs ? datasets, extent ...

I tried to replicate with Maumere tsunami raster on population raster : 
![capture d ecran 2017-03-09 a 08 34 13](https://cloud.githubusercontent.com/assets/1609292/23740325/60f732da-04a3-11e7-8bf3-08cbf68f201b.png)

I tried with EQ raster on population raster too : 
![capture d ecran 2017-03-09 a 08 37 29](https://cloud.githubusercontent.com/assets/1609292/23740403/b05c1a2a-04a3-11e7-8933-ae0e28394281.png)

Everytime I use the biggest area as possible. I can't replicate.

**Comment by Charlotte-Morgan** [comment]
hi @ivanbusthomi - can you please replicate?
use Sunda WorldPop 2015, Sunda Slab EQ
share the world pop if necessary, I already shared hazard data yesterday.
Thanks

**Comment by Charlotte-Morgan** [comment]
my negative is only in the general report not the detailed report

**Comment by samnawi** [comment]
Hi @Gustry I got the same problem with @Charlotte-Morgan when I doing analysis for whole area.
![image](https://cloud.githubusercontent.com/assets/8871958/23744566/4b225cca-04e7-11e7-91e7-1ed216d96d4b.png)

But, when I tried in the smaller area, the result is fine
![image](https://cloud.githubusercontent.com/assets/8871958/23744579/5a60ccc6-04e7-11e7-8893-f11da1687cd9.png)

**Comment by ivanbusthomi** [comment]
Hi @Gustry 
I can replicate the problem. I already put the exposure data in `Real_world_data` BtSync, under `Sunda` folder

**Comment by Gustry** [comment]
Thanks, I will check

**Comment by Gustry** [comment]
I could replicate the issue. We need a big dataset for this one.

The number of displaced : 47 122 267
In the BNPB profile, we distribute 67 L of clean water for each displaced person : 
67 * 47 122 267 = 3 157 191 916

There is somehow a maximum of the number. We use memory layer and geojson layer during the process.

Solutions : 
* Switch to m³ in the profile, the easiest one for us.
* Switch automatically to m³ if needed but we can't do that now, InaSAFE is not aware about the original unit. We have to update the minimum needs to be able to set the unit from a combobox ans we will be able to know which units are compatible each other.
* Increase the field: QLonglong with length 1. I did that for now in PR #4044. But it's a temporary solution until we have a bigger dataset with more people displaced.

**Comment by Charlotte-Morgan** [comment]
please check - I think this is fixed?

**Comment by timlinux** [comment]
@gustry can you update on the status of this?

**Comment by ismailsunni** [comment]
Checking the max value of QLongLong is 2^64 - 1. So, even if all of human affected (7*10^9), we will still be safe.

**Comment by Gustry** [comment]
Yes, but I'm not sure QLongLong is supported then when we create layers (memory layer or geojson layer or any vector layer)

**Comment by Gustry** [comment]
I already put my solutions in the post : https://github.com/inasafe/inasafe/issues/4039
Nothing more for now

**Comment by Charlotte-Morgan** [comment]
@samnawi - can you please see if this is still a problem?

**Comment by Charlotte-Morgan** [comment]
@felix-yew  - can you please check this one too?

**Comment by ismailsunni** [comment]
No matter what we do, there will be a number that exceed the maximum number (e.g. when the minimum needs have a big number). So, if for the scenario in this ticket, the solution from @Gustry works, I think we can close this ticket.

OOT : This also happened to YouTube (Google), when Psy - Gangnam Style exceeded the maximum number that they set.
